src/main/java/org/opentripplanner/routing/algorithm/raptoradapter/transit/request/RouteRequestTransitDataProviderFilter.java

```diff
@@ -197,6 +216,20 @@ private static boolean routeIsBanned(
       }
     }
 
+    var groupOfRoutesIDs = route
+      .getGroupsOfRoutes()
+      .stream()
+      .map(AbstractTransitEntity::getId)
+      .toList();
+
+    if (!bannedGroupsOfRoutes.isEmpty()) {
+      for (var id : groupOfRoutesIDs) {
```

Comment:
I believe Collections.disjoint is the fastest way to compare if all elements are not in the other set.

Reply:
You are right, I changed the implementation